src/base.jl

```diff
@@ -221,26 +221,32 @@ end
 # need each PowerSystemType to store a UUID.
 # GitHub issue #203
 
+# Const Definitions
 
-"""Returns an iterable of components from the System. T can be concrete or abstract.
+
+"""Defines the Iterator to contain variables of type Component.
+"""
+const ComponentIterator{T} = Base.Iterators.Flatten{Array{Array{T,1},1}} where {T <: Component}
+
+"""Returns a ComponentIterator{T} from the System. T can be concrete or abstract.
 
 # Examples
 ```julia
 devices = PowerSystems.get_components(ThermalDispatch, system)
 generators = PowerSystems.get_components(Generator, system)
 ```
 """
-function get_components(::Type{T}, sys::ConcreteSystem) where {T <: Component}
+function get_components(::Type{T}, sys::ConcreteSystem)::ComponentIterator{T} where {T <: Component}
     if isconcretetype(T)
         components = get(sys.components, T, nothing)
         if isnothing(components)
-            return Iterators.flatten([[]])
+            return ComponentIterator{T}(Vector{T}())
```

Comment:
I just noticed that you've changed this section again.  There are three cases where you create a ComponentIterator.  In the first you explicitly create a Vector{T} instead of [].  In the next two you create the object from [...]. These should be consistent.  If using [] works, then you can just delete Vector.  Also, your type definition uses Array{T, 1} instead of Vector.  These should also be consistent.  I think we should always use Vector{T} instead of Array{T, 1}.

Reply:
If I don't specify the type inside of the ComponentIterator then it returns a type `Iterators.flatten[Array{Any,1}]` which made the tests fail. This syntax will always return an iterator with the correct Array Type parametrization even if it needs to be empty instead of an empty Array parametrized with Any. 

I can change the ComponentIterator to use Vector that is not an issue.

Reply:
Ok, makes sense.  In cases 2 and 3 does it still do the right thing if the vectors are empty?

Reply:
Yes, the return is fine